typescript/sdk/src/core/HyperlaneCore.ts

```diff
@@ -85,6 +89,50 @@ export class HyperlaneCore extends HyperlaneApp<CoreFactories> {
     return this.multiProvider.getChainName(message.parsed.destination);
   }
 
+  getRecipientIsmAddress(message: DispatchedMessage): Promise<Address> {
+    const destinationMailbox = this.contractsMap[this.getDestination(message)];
+    const ethAddress = bytes32ToAddress(message.parsed.recipient);
+    return destinationMailbox.mailbox.recipientIsm(ethAddress);
+  }
+
+  async getRecipientIsmConfig(
+    message: DispatchedMessage,
+  ): Promise<WithAddress<IsmConfig>> {
+    const destinationChain = this.getDestination(message);
+    const ismReader = new EvmIsmReader(this.multiProvider, destinationChain);
+    const address = await this.getRecipientIsmAddress(message);
+    return ismReader.deriveIsmConfig(address);
+  }
+
+  async relayMessage(
+    message: DispatchedMessage,
+  ): Promise<ethers.ContractReceipt> {
+    const destinationChain = this.getDestination(message);
+    const mailbox = this.contractsMap[destinationChain].mailbox;
+    const ismConfig = await this.getRecipientIsmConfig(message);
+    if (typeof ismConfig === 'string') {
+      throw new Error(
+        `No ISM config found for recipient ${message.parsed.recipient}`,
+      );
+    }
```

Comment:
Does a string config mean it's an address? The error message is confusing me a bit here

Reply:
does `getRecipientIsmConfig` actually return a type `string`? I would've thought that with the type `WithAddress<IsmConfig>`, we'd get back an object instead of just the string i.e.
```
{
    string: '0xsdsd',
    address: '0xsdsd',
}
```

curious if you hit a problem when running that led to this check

Reply:
changed your return type to `Exclude`